src/sources/util/http/prelude.rs

```diff
@@ -45,6 +47,7 @@ pub trait HttpSource: Clone + Send + Sync + 'static {
         self,
         address: SocketAddr,
         path: &str,
+        http_method: Method,
```

Comment:
I would have this take the `HttpMethod` type you defined above. That way the below `match` could be an exhaustive match on the enum rather than following back to `POST` on line 71 (which would never actually be hit in practice since we verify earlier that the incoming method is one of a defined set).